models/content_test.go

```diff
@@ -0,0 +1,125 @@
+		content := models.Content{
+			State: models.StatePublished.String(),
+		}
+		err := content.Validate()
+		So(err, ShouldBeNil)
+	})
+}
+
+func TestContentValidateTransitionFrom(t *testing.T) {
+	Convey("Given an existing content in an created state", t, func() {
+		existing := &models.Content{
+			State: models.StateCreated.String(),
+		}
+
+		Convey("When we try to transition to an completed state", func() {
+			content := &models.Content{
+				State: models.StateCompleted.String(),
+			}
+			err := content.ValidateTransitionFrom(existing)
+			So(err, ShouldBeNil)
+		})
+
+		Convey("When we try to transition to a published state", func() {
+			content := &models.Content{
+				State: models.StatePublished.String(),
+			}
+			err := content.ValidateTransitionFrom(existing)
+			So(err, ShouldResemble, apierrors.ErrTopicStateTransitionNotAllowed)
+		})
+	})
+
+	Convey("Given an existing content in a Completed state", t, func() {
+		existing := &models.Content{
+			State: models.StateCompleted.String(),
+		}
+
+		Convey("When we try to transition to a published state", func() {
+			content := &models.Content{
+				State: models.StatePublished.String(),
+			}
+			err := content.ValidateTransitionFrom(existing)
+			So(err, ShouldBeNil)
+		})
+
+		Convey("When we try to transition to an created state", func() {
+			content := &models.Content{
+				State: models.StateCreated.String(),
+			}
+			err := content.ValidateTransitionFrom(existing)
+			So(err, ShouldResemble, apierrors.ErrTopicStateTransitionNotAllowed)
+		})
+	})
+}
+
+func TestContentStateTransitionAllowed(t *testing.T) {
+	Convey("Given an content in created state", t, func() {
+		content := models.Content{
+			State: models.StateCreated.String(),
+		}
+		contentValidateTransitionsToCreated(content)
+	})
+
+	Convey("Given an content with a wrong state value, then no transition is allowed", t, func() {
+		content := models.Content{State: "wrong"}
+		contentValidateTransitionsToCreated(content)
+	})
+
+	Convey("Given an content without state, then created state is assumed when checking for transitions", t, func() {
+		content := models.Content{}
+		contentValidateTransitionsToCreated(content)
+	})
+}
+
+// validateTransitionsToCreated validates that the provided content can transition to created state,
+// and not to any forbidden of invalid state
+func contentValidateTransitionsToCreated(content models.Content) {
+	Convey("Then an allowed transition is successfully checked", func() {
+		So(content.StateTransitionAllowed(models.StateCompleted.String()), ShouldBeTrue)
+	})
+	Convey("Then a forbidden transition to a valid state is not allowed", func() {
+		So(content.StateTransitionAllowed(models.StatePublished.String()), ShouldBeFalse)
+	})
+	Convey("Then a transition to an invalid state is not allowed", func() {
+		So(content.StateTransitionAllowed("wrong"), ShouldBeFalse)
+	})
+}
+
+//!!! as the content functionality grows, add more tests to cover added state transitions.
```

Comment:
I'm not sure these comments add value particularly. It should be a standard expectation that unit tests are added as functionality is changed, theres no need to call it out.

Reply:
The same with the next comment, I'm not sure if its a reminder to yourself in the next few days to make this more generic, or a reminder to future devs to keep the split that you've already created. Its causing more confusion than the value it adds to me. If they're all reminders to yourself I'd suggest using git patches to apply smaller subsets of clean code so you can keep your notes locally without having worrying if they're clear to others.

Reply:
removed